private/buf/cmd/buf/command/curl/curl.go

```diff
@@ -497,6 +545,117 @@ func (f *flags) validate(isSecure bool) error {
 	return nil
 }
 
+func (f *flags) determineCredentials(ctx context.Context, container app.Container, host string) (string, error) {
+	if f.UserCreds != "" {
+		// this flag overrides any netrc-related flags
+		var username, password string
+		pos := strings.IndexByte(f.UserCreds, ':')
+		if pos == -1 {
+			username = f.UserCreds
+			var err error
+			password, err = promptForPassword(ctx, container, fmt.Sprintf("Enter host password for user %q:", username))
+			if err != nil {
+				return "", fmt.Errorf("could not prompt for password: %w", err)
+			}
+		} else {
+			username = f.UserCreds[:pos]
+			password = f.UserCreds[pos+1:]
+		}
+		return basicAuth(username, password), nil
+	}
+
+	// process netrc-related flags
+	netrcFile := f.NetrcFile
+	if netrcFile == "" {
+		if !f.UseNetrc {
+			// no netrc file usage, so no creds
+			return "", nil
+		}
+		var err error
+		netrcFile, err = netrc.GetFilePath(container)
+		if err != nil {
+			return "", fmt.Errorf("could not determine path to .netrc file: %w", err)
+		}
+	}
+	if _, err := os.Stat(netrcFile); err != nil {
+		if !strings.Contains(err.Error(), netrcFile) {
+			// make sure error message contains path to file
+			return "", fmt.Errorf("could not read file: %s: %w", netrcFile, err)
+		}
+		return "", fmt.Errorf("could not read file: %w", err)
+	}
+	// NB: can't use buf/private/pkg/netrc for this because it only supports reading from $HOME/.netrc
```

Comment:
We should just extend pkg/netrc then, it's meant to be a wrapper so we don't expose the external lib throughout the codebase

Reply:
Fair. And it was trivial to just expose an existing unexported method. Done.